src/types/AuditLogEvent.ts

```diff
@@ -0,0 +1,42 @@
+export default class AuditLogEvent {
+  private constructor(private code: string, private description: string) {}
+
+  public static readonly loggedIn = new AuditLogEvent("user-logged-in", "User logged in")
+
+  public static readonly changedPassword = new AuditLogEvent("user-password.changed", "Change password")
+
+  public static readonly deletedUser = new AuditLogEvent("user-deletion.deleted", "Delete user")
+
+  public static readonly resetPassword = new AuditLogEvent("user-password.reset", "Reset password")
+
+  public static readonly createdUser = new AuditLogEvent("user-creation.created", "Create user")
+
+  public static readonly failedCreatingNewEmailWhileCreatingUser = new AuditLogEvent(
+    "user-creation.failed-email",
+    "Error creating new user email"
+  )
+
+  public static readonly failedNotificationWhileCreatingUser = new AuditLogEvent(
+    "user-creation.failed-notification",
+    "Error sending notification email of new user creation"
+  )
+
+  public static readonly failedToEmailUserWhileCreatingUser = new AuditLogEvent(
+    "user-creation.failed-emailing-user",
+    "Error sending email to new user"
+  )
+
+  public static readonly editedUser = new AuditLogEvent("user-edit.updated", "Edit user")
```

Comment:
This is looking much better, but I think we could probably tweak the codes a little to match the format of the ones we already generate in Bichard. Typically they are `<noun>.<verb>` - such as `hearing-outcome.received` or `exceptions.generated`.

Here are some suggestions:

| Event Type | Event Code |
|---|---|
| User logged in | `user.logged-in` |
| Password updated | `user.password.updated` |
| Password reset | `user.password.reset` |
| User deleted | `user.deleted` |
| User created | `user.created` |

I don't know if the "Error creating new user email", "Error sending notification email of new user creation" and "Error sending email to new user" should be audit logs at all - they're more application-level error logs, not something we need to keep for posterity and auditing purposes.

I also think we probably need to look through the user-service and make sure we're audit logging all of the actions users can make - like updating the triggers or details on themselves or other users. But that could be a separate PR/ticket.

Reply:
Thanks for the event codes 🙌. It looks much cleaner and more transparent.
I agree, we probably need to log the failure logs as before in console so logs will be picked by logstash and we can see them in OpenSearch if we need to debug or investigate an issue.

Reply:
👍 Agree with Jamie's points but looks like you're already on it!